daemon/transmission-daemon.service

```diff
@@ -8,10 +8,34 @@ User=transmission
 Type=notify
 ExecStart=/usr/bin/transmission-daemon -f --log-level=error
 ExecReload=/bin/kill -s HUP $MAINPID
+
+# Hardening
 NoNewPrivileges=true
 MemoryDenyWriteExecute=true
 ProtectSystem=true
 PrivateTmp=true
+PrivateDevices=true
+DevicePolicy=closed
+ProtectClock=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+ProtectKernelModules=true
+ProtectHostname=true
+ProtectHome=true
```

Comment:
It was brought to my attention that this could, in theory, break people's setups if they allow the `transmission` user to access their `$HOME`. IMO, that is a crazy scenario that is wholly inadvisable and shouldn't be supported. But it still *could* be a breaking change.

Reply:
Let's not ProtectHome by default.
People who'll want to achieve the same thing can just disallow access to $HOME. But for others, Transmission would be a non-functioning software and would result in frustrations and issues if we ProtectHome.

Reply:
Amen

Reply:
To be clear this is for the daemon only, running as a system service only. Even without this change, as the service is currently written a user would need to manually intervene to make writing to `$HOME` possible. So in the default configuration this breaks nothing.

Reply:
I like the idea, but that sounds like it might involve a lot of work, no?

Reply:
It's quite easy, #854 was mostly there, but not quite correct. Certain directives don't work, and in particular the user manager has no access to `network-online.target`, see https://github.com/systemd/systemd/issues/28762. So it wouldn't be perfect, but transmission technically doesn't need internet in order to start.